source/localizable/tutorial/subroutes.md

```diff
@@ -0,0 +1,425 @@
+
+```app/routes/rentals.hbs{-2,-3,-4}
+export default Ember.Route.extend({
+  model() {
+    return this.store.findAll('rental');
+  }
+});
+```
+
+```app/routes/rentals/index.js{+2,+3,+4}
+export default Ember.Route.extend({
+  model() {
+    return this.store.findAll('rental');
+  }
+});
+```
+
+Now that we are returning all of our rentals to the nested route's model, we will also move the rental list markup from our main route template to our nested route index template.
+
+```app/templates/rentals.hbs{-9,-10,-11,-12,-13,-14,-15,-16,-17}
+<div class="jumbo">
+  <div class="right tomster"></div>
+  <h2>Welcome!</h2>
+  <p>We hope you find exactly what you're looking for in a place to stay.</p>
+  {{#link-to 'about' class="button"}}
+    About Us
+  {{/link-to}}
+</div>
+{{#list-filter
+   filter=(action 'filterByCity')
+   as |rentals|}}
+  <ul class="results">
+    {{#each rentals as |rentalUnit|}}
+      <li>{{rental-listing rental=rentalUnit}}</li>
+    {{/each}}
+  </ul>
+{{/list-filter}}
+{{outlet}}
+```
+
+```app/templates/rentals/index.hbs{+1,+2,+3,+4,+5,+6,+7,+8,+9}
+{{#list-filter
+   filter=(action 'filterByCity')
+   as |rentals|}}
+  <ul class="results">
+    {{#each rentals as |rentalUnit|}}
+      <li>{{rental-listing rental=rentalUnit}}</li>
+    {{/each}}
+  </ul>
+{{/list-filter}}
+{{outlet}}
+```
+
+Finally, we need to make our controller that has our filter action available to the new nested index route.
+Instead of moving the whole controller file over to `app/controller/rentals/index.js` from `app/controller/rentals.js`, we'll just take advantage of the route's `controllerName` property to just point at our existing `rentals` controller.
+
+```app/routes/rentals/index.js{+2}
+export default Ember.Route.extend({
+  controllerName: 'rentals',
+  model() {
+    return this.store.findAll('rental');
+  }
+});
+```
+
+## Setting up Data for the Nested Detail Route
+
+Next, we will want to create a sub-route that will list information for a specific rental.
+To do this, we will need to update a couple of files.
+To find a specific rental, we will want to use Ember Data's `findRecord` function [(see "Finding Records" for more details)](../../models/finding-records/).
+The `findRecord` function requires that we search by a unique key.
+
+While on the `show` route, we will also want to show additional information about our specific rental.
+
+In order to do this, we need to modify the Mirage `config.js` file.
+If you need a refresher on how Mirage works, go back to the [Installing Addons section](../installing-addons)
+We will add a new route handler to mirage to handle when the http request comes in to fetch a rental.
+
+```mirage/config.js{+57,+58,+59,+60}
+export default function() {
+  this.namespace = '/api';
+
+  let rentals = [
+    {
+      type: 'rentals',
+      id: 'grand-old-mansion',
```

Comment:
Are we switching to string ids here?  It's quite elegant if we can do it, but we'll want to indicate that we changed these, as our current mirage conf uses numeric ids instead